src/util/write-payload.ts

```diff
@@ -185,10 +187,14 @@ export class WritePayload<T extends SpraypaintBase> {
     const wp = new WritePayload(model, nested, idOnly)
     const relatedJSON = wp.asJSON().data
 
-    this._pushInclude(relatedJSON)
+    if(!(!model.isPersisted && model.isMarkedForDestruction)) {
```

Comment:
These particular boolean combinations are a bit hard to parse, particularly when repeated a bunch throughout this file.  How about extracting into a utility method that accepts a model and returns a boolean, which will allow a better landing spot for a comment explaining this context.

Reply:
Added a private method to write-payloads as suggested.